geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/tables/AttributeTable.scala

```diff
@@ -164,20 +167,22 @@ object AttributeTable extends GeoMesaTable with Logging {
   /**
    * Lexicographically encode the value. Collections will return multiple rows, one for each entry.
    *
-   * @param valueOption
+   * @param value
    * @param descriptor
    * @return
    */
-  def encode(valueOption: Option[Any], descriptor: AttributeDescriptor): Seq[String] = {
-    val value = valueOption.getOrElse(nullString)
-    if (descriptor.isCollection) {
+  def encode(value: Any, descriptor: AttributeDescriptor): Seq[String] = {
+    if (value == null) {
+      Seq.empty
+    } else if (descriptor.isCollection) {
       // encode each value into a separate row
-      value.asInstanceOf[JCollection[_]].asScala.toSeq.map(Option(_).getOrElse(nullString)).map(typeEncode)
+      value.asInstanceOf[JCollection[_]].toSeq.flatMap(Option(_).map(typeEncode).filterNot(_.isEmpty))
```

Comment:
I feel like there is some minor cost involved in the implicit conversion to a GenTraversableOnce with .nonEmpty, but I can't really confirm that... Any thoughts?